tests/build/build_templates.go

```diff
@@ -602,3 +615,41 @@ func getImageWithDigest(c *framework.ControllerHub, componentName, applicationNa
 	}
 	return fmt.Sprintf("%s@%s", url, digest), nil
 }
+
+func updateBuildPipelineSelector(kubeadminClient *framework.ControllerHub, testNamespace string) error {
+	//trueBool := true
+	customSourceBuildBundle := os.Getenv(constants.CUSTOM_SOURCE_BUILD_PIPELINE_BUNDLE_ENV)
+	Expect(customSourceBuildBundle).ShouldNot(BeEmpty())
+	if customSourceBuildBundle == "" {
+		return fmt.Errorf("source build bundle is empty")
+	}
+	bpsObj := &buildservice.BuildPipelineSelector{}
+	err := kubeadminClient.CommonController.KubeRest().Get(context.Background(), types.NamespacedName{Name: "build-pipeline-selector", Namespace: testNamespace}, bpsObj)
+	if err != nil {
+		return fmt.Errorf("error while getting build-pipeline-selector: %v", err)
+	}
+	found := false
+	for i := range bpsObj.Spec.Selectors {
+		selector := &bpsObj.Spec.Selectors[i]
+		if selector.Name == "Docker build" {
+			params := bpsObj.Spec.Selectors[i].PipelineRef.Params
+			for _, param := range params {
+				if param.Name == "bundle" {
+					found = true
+					break
+				}
+			}
+		}
+		if found {
+			selector.PipelineRef.Params[i].Value = *tektonpipeline.NewStructuredValues(customSourceBuildBundle)
+			break
+		}
+	}
+	GinkgoWriter.Printf("BPS AFTER UPDATE: %v\n", bpsObj.Spec.Selectors)
+
+	err = kubeadminClient.CommonController.KubeRest().Update(context.Background(), bpsObj)
+	if err != nil {
+		return fmt.Errorf("error while updating build pipeline selector: %v", err)
+	}
+	return nil
```

Comment:
Probably we should also handle the case of the "Docker build" pipeline is not found.

Reply:
Thanks. Updated now.